src/screens/Analytics.tsx

```diff
@@ -1,20 +1,220 @@
 import { Container, Paper, Tab, Tabs, Stack } from "@mui/material";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import { useLocation, useParams } from "react-router";
+import { filterByStreamId } from "../adapters/stream";
 import { ComposedTwoAreasChart } from "../component/ComposedTwoAreasChart";
 import { ComposedTwoYAxisChart } from "../component/ComposedTwoYAxisChart";
 import { HardwareInformationTable } from "../component/HardwareInformationTable";
-import { Map } from "../component/Map";
+import { LowAudioBitrateEvent, LowBitrateEvent, Map } from "../component/Map";
 import { SimpleStatsTable } from "../component/SimpleStatsTable";
 import { StackAreasChart } from "../component/StackAreasChart";
 import { StreamHealthTable } from "../component/StreamHealthTable";
+import { ModemModel } from "../models/modem";
+import { StreamDatapointModel } from "../models/stream_datepoint";
+import { StreamModel } from "../models/stream_sessions";
 
 export function Analytics() {
   const [value, setValue] = useState(0);
 
+  const { streamId }: { streamId: string } = useParams();
+  const location = useLocation();
+  const streamModel = location.state as StreamModel;
+  const videoCodec = streamModel.videoCodec;
+  const audioCodec = streamModel.audioCodec;
+  const videoResolution = streamModel.videoResolution;
+
+  const [streamData, setStreamData] = useState<StreamDatapointModel[]>([]);
+  const [displayData, setDisplayData] = useState({
+    minBitrate: 0,
+    maxBitrate: 0,
+    avgBitrate: 0,
+    minFPS: 0,
+    maxFPS: 0,
+    avgFPS: 0,
+    minAudioBitrate: 0,
+    maxAudioBitrate: 0,
+    avgAudioBitrate: 0,
+    modemKeys: [""],
+    aggregateUpstreamData: [],
+    aggregateDownstreamData: [],
+    individualModemsData: [[{ upstream: 0, downstream: 0 }]],
+    lowBitrateEvents: [{ x: 0, y: 0, bitrate: 0, timestamp: new Date() }],
+    lowAudioBitrateEvents: [
+      { x: 0, y: 0, audioBitrate: 0, timestamp: new Date() },
+    ],
+    healthCellMetaData: [
+      { x: 0, y: 0, bitrate: 0, timestamp: new Date() },
+      { x: 0, y: 0, audioBitrate: 0, timestamp: new Date() },
+    ],
+  });
+
+  useEffect(() => {
+    async function loadStreamData(streamId: string) {
```

Comment:
Would it be better to break this function into smaller ones so that its easier to follow?